spec/ParseServer.spec.js

```diff
@@ -114,3 +117,53 @@ describe('Server Url Checks', () => {
     });
   });
 });
+
+describe('Server Shutdown', () => {
+  let quitSpy;
+  let databaseAdapter;
+  let filesAdapter;
+
+  beforeEach(() => {
+    // Mock redis
+    quitSpy = jasmine.createSpy('quit'); // Called when connection is closed
+    const createClient = jasmine.createSpy('createClient').and.returnValue({
+      quit: quitSpy,
+    });
+    jasmine.mockLibrary('redis', 'createClient', createClient);
+
+    const mongoURI = 'mongodb://localhost:27017/parseServerMongoAdapterTestDatabase';
+    const postgresURI = 'postgres://localhost:5432/parse_server_postgres_adapter_test_database';
+
+    const isPostgres = process.env.PARSE_SERVER_TEST_DB === 'postgres';
+    databaseAdapter = isPostgres
+      ? new PostgresStorageAdapter({
```

Comment:
Not sure if instantiating a new DB adapter with custom settings outside of the test framework could have any side effects regarding cleanup, etc.

@dplewis ?